apps/test/unit/lib/levelbuilder/CourseOfferingEditorTest.jsx

```diff
@@ -0,0 +1,172 @@
+import {expect} from '../../../util/reconfiguredChai';
+import React from 'react';
+import {mount} from 'enzyme';
+import CourseOfferingEditor from '@cdo/apps/lib/levelbuilder/CourseOfferingEditor';
+import sinon from 'sinon';
+import * as utils from '@cdo/apps/utils';
+
+describe('CourseOfferingEditor', () => {
+  let defaultProps;
+
+  beforeEach(() => {
+    defaultProps = {
+      initialCourseOffering: {
+        key: 'test-course-offering',
+        is_featured: false,
+        category: 'Other',
+        display_name: 'Course Offering 1'
+      }
+    };
+  });
+
+  describe('Saving Course Offering Editor', () => {
+    let clock, server;
+
+    beforeEach(() => {
+      server = sinon.fakeServer.create();
+      sinon.stub(utils, 'navigateToHref');
+    });
+
+    afterEach(() => {
+      if (clock) {
+        clock.restore();
+        clock = undefined;
+      }
+      server.restore();
+      utils.navigateToHref.restore();
+    });
+
+    it('can save and keep editing', () => {
+      const wrapper = mount(<CourseOfferingEditor {...defaultProps} />);
+
+      let returnData = {
+        key: 'test-course-offering',
+        display_name: 'Course Offering 2',
+        category: 'Full Courses',
+        is_featured: true
+      };
+      server.respondWith('PUT', '/course_offerings/test-course-offering', [
+        200,
+        {'Content-Type': 'application/json'},
+        JSON.stringify(returnData)
+      ]);
+
+      const saveBar = wrapper.find('SaveBar');
+
+      const saveAndKeepEditingButton = saveBar.find('button').at(1);
+      expect(saveAndKeepEditingButton.contains('Save and Keep Editing')).to.be
+        .true;
+      saveAndKeepEditingButton.simulate('click');
+
+      // check the the spinner is showing
```

Comment:
nit: I'm not sure that `CourseOfferingEditor` cares if the spinner is showing. The logic belongs to `SaveBar`

Longer ramble: I find it difficult to separate the concerns of `SaveBar` with the concerns of the editor (I had this problem with `ProgrammingExpressionEditor` as well). The test needs access to `SaveBar` to initiate saving but after that, I'm not sure what needs to be checked. Maybe the spinner being there should be checked, maybe it doesn't matter. 🤷‍♀️

Reply:
Ya I also struggled. I have found figuring out how to test with the SaveBar complicated